expand/expand.go

```diff
@@ -375,10 +375,8 @@ func Fields(cfg *Config, words ...*syntax.Word) ([]string, error) {
 	fields := make([]string, 0, len(words))
 	dir := cfg.envGet("PWD")
 	for _, word := range words {
-		afterBraces := []*syntax.Word{word}
-		if syntax.SplitBraces(word) {
-			afterBraces = Braces(word)
-		}
+		syntax.SplitBraces(word)
+		afterBraces := Braces(word)
```

Comment:
How about:

```
word := *word // make a copy, since SplitBraces replaces the Parts slice
if syntax.SplitBraces(&word) {
    afterBraces = Braces(&word)
}
```

In hindsight, making SplitBraces an "in-place" API might have been a bad idea, because of subtle bugs like this. The idea was to not need to allocate for the base case of there being zero brace expressions, which I think is very common in words.

In any case, my version is perhaps a bit more complex, but I don't want Fields to be modifying its parameters in-place. The API isn't meant to do that, and it was never documented that way. It also keeps the property of only calling `Braces` when we need to.